client/control/components/GroupMenu.js

```diff
@@ -0,0 +1,97 @@
+import React, { PropTypes as pt } from 'react';
+
+/**
+ * Simple menu component which displays groups of items
+ * under section headers/labels.
+ */
+export default class GroupMenu extends React.Component {
+  static propTypes = {
+    data: pt.array.isRequired,
+    onItemSelect: pt.func.isRequired,
+    searchText: pt.string,
+  };
+
+  /**
+   * Creates a click handler based on the
+   * group/option passed into it, basically
+   * an event handler factory.
+   *
+   * @param  {Object} group  Group which clicked option belongs to
+   * @param  {Object} option Option which was clicked
+   * @return {Function}      Wrapped handler
+   */
+  handleItemClick(group, option) {
+    return () => this.props.onItemSelect(group, option);
```

Comment:
Oh, you're right. I forgot about the perf thing. It also makes us re-render more than we should.

I still think the naming issue stands though.